pkg/client/optimizely_config_test.go

```diff
@@ -0,0 +1,81 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"testing"
+
+	"github.com/optimizely/go-sdk/pkg/config"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type OptimizelyConfigTestSuite struct {
+	suite.Suite
+	optimizelyClient         *OptimizelyClient
+	expectedOptimizelyConfig OptimizelyConfig
+}
+
+func (s *OptimizelyConfigTestSuite) SetupTest() {
+	dataFileName := "testdata/optimizely_config_datafile.json"
+	datafile, err := ioutil.ReadFile(dataFileName)
+	if err != nil {
+		s.Fail("error opening file:" + dataFileName)
+	}
+
+	projectConfigManager, e := config.NewStaticProjectConfigManagerFromPayload(datafile)
+	if e != nil {
+		s.Fail("error constructing config manager")
+	}
+
+	outputFileName := "testdata/optimizely_config_expected.json"
+	expectedOutput, er := ioutil.ReadFile(outputFileName)
```

Comment:
I understand this file read and map guarantees that all data fields are public, so no need for extra access control validation. If it's intention, add comments to clarify it.

Reply:
comments provided